spec/03-plugins/24-rate-limiting/05-integration_spec.lua

```diff
@@ -0,0 +1,177 @@
+local helpers = require "spec.helpers"
+local redis = require "resty.redis"
+
+local REDIS_HOST = "127.0.0.1"
+local REDIS_PORT = 6379
+local REDIS_PASSWORD = ""
+local REDIS_DATABASE = 1
+local REDIS_DATABASE_SECOND = 0
+
+local SLEEP_TIME = 1
+
+local function flush_redis(db)
+  local red = redis:new()
+  red:set_timeout(2000)
+  local ok, err = red:connect(REDIS_HOST, REDIS_PORT)
+  if not ok then
+    error("failed to connect to Redis: " .. err)
+  end
+
+  if REDIS_PASSWORD and REDIS_PASSWORD ~= "" then
+    local ok, err = red:auth(REDIS_PASSWORD)
+    if not ok then
+      error("failed to connect to Redis: " .. err)
+    end
+  end
+
+  local ok, err = red:select(db)
+  if not ok then
+    error("failed to change Redis database: " .. err)
+  end
+
+  red:flushall()
+  red:close()
+end
+
+describe("Plugin: rate-limiting (integration)", function()
+
+  local client
+
+  setup(function()
+    helpers.dao:drop_schema()
+    helpers.run_migrations()
+  end)
+
+  teardown(function()
+    flush_redis(REDIS_DATABASE)
+    flush_redis(REDIS_DATABASE_SECOND)
```

Comment:
I'm not sure how fast those operations are, but if we can avoid it, we should. There is no point in cleaning up behind ourselves in our tests, since they always run in a test instance and because it is actually beneficial to preserve the tests results, for a potential investigation around a failing test. We do this a lot. Plus the `setup()` hooks performs the same operations as well for the next run of this test suite.